sdlib/d/gc/page.d

```diff
@@ -728,6 +731,44 @@ private:
 					occupancyMask |= newOccupancy;
 					count += popCount(oldOccupancy ^ newOccupancy);
 
+					if (!ec.supportsMetadata) {
+						e.slabData.rawContent[i] = newOccupancy;
```

Comment:
This is duplicated, so really not the ideal setup. generally, this code has too many nesting levels, it clearly needs a refactoring. incidentally, part of the refactoring might also apply to sparse slabs.

Reply:
Switched to a scope(exit), which works nicely, and saves another indent.